src/domainobjectfactories/creatable.py

```diff
@@ -501,6 +501,40 @@ def create_return_type(self):
 
     # THESE ARE NON-GENERATING, UTILITY METHODS USED WHERE NECESSARY #
 
+    def get_random_record_with_valid_attribute(
+            self, table_name, attribute_to_validate, invalid_values
+    ):
+        """ returns a random record from a specified database table, subject
+        to the constraint that a specified attribute not have a value in a
+        specified list of invalid values
+
+        Parameters
+        ----------
+        table_name : String
+            Name of the database table to select the valid record from
+        attribute_to_validate: String
+            Attribute key for which the value will determine if the record is valid
+        invalid_values: List
+            List of 1 or more invalid values for the attribute given by the
+            attribute_to_validate parameter. Only records with values for
+            that attribute not in this list will be selected in the database
+            query.
+        """
+        # turn the invalid_values list into a string with format:
+        # "('value1', 'value2', etc..)"
+        invalid_values = \
+            ', '.join([f"\'{value}\'" for value in invalid_values])
```

Comment:
I like this logic but I would rather than sql command itself was in sqlite_database.py, as that's the pattern we're using with methods like createable.retrieve_records and sqlte_database.retrieve